backend/devices/bitbox02/keystore.go

```diff
@@ -0,0 +1,116 @@
+// Copyright 2018 Shift Devices AG
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package bitbox02
+
+import (
+	"github.com/btcsuite/btcutil/hdkeychain"
+	"github.com/digitalbitbox/bitbox-wallet-app/backend/coins/btc"
+	coinpkg "github.com/digitalbitbox/bitbox-wallet-app/backend/coins/coin"
+	"github.com/digitalbitbox/bitbox-wallet-app/backend/coins/eth"
+	"github.com/digitalbitbox/bitbox-wallet-app/backend/devices/bitbox02/messages"
+	"github.com/digitalbitbox/bitbox-wallet-app/backend/signing"
+	"github.com/digitalbitbox/bitbox-wallet-app/util/errp"
+	"github.com/sirupsen/logrus"
+)
+
+var msgCoinMap = map[string]messages.BTCCoin{
+	"btc":  messages.BTCCoin_BTC,
+	"tbtc": messages.BTCCoin_TBTC,
+	"ltc":  messages.BTCCoin_LTC,
+	"tltc": messages.BTCCoin_TLTC,
+}
+
+type keystore struct {
+	device        *Device
+	configuration *signing.Configuration
+	cosignerIndex int
+	log           *logrus.Entry
+}
+
+// CosignerIndex implements keystore.Keystore.
+func (keystore *keystore) CosignerIndex() int {
+	return keystore.cosignerIndex
+}
+
+// HasSecureOutput implements keystore.Keystore.
+func (keystore *keystore) HasSecureOutput(configuration *signing.Configuration, coin coinpkg.Coin) (bool, error) {
+	_, ok := msgCoinMap[coin.Code()]
+	return ok, nil
+}
+
+// OutputAddress implements keystore.Keystore.
+func (keystore *keystore) OutputAddress(
+	configuration *signing.Configuration, coin coinpkg.Coin) error {
+	hasSecureOutput, err := keystore.HasSecureOutput(configuration, coin)
+	if err != nil {
+		return err
+	}
+	if !hasSecureOutput {
+		panic("HasSecureOutput must be true")
+	}
+	msgCoin, ok := msgCoinMap[coin.Code()]
+	if !ok {
+		panic("unsupported coin")
```

Comment:
Doesn't `HasSecureOutput()` already check the same thing with `_, ok := msgCoinMap[coin.Code()]`?
So if the above condition (`if !hasSecureOutput`) passes, this check should also always pass. 
Maybe these could be combined into one step? Something like `coinSupportedAndHasSecureOutput`, but with better name :D

Reply:
good catch, removing those lines.